File: frontend/src/types/criteria.ts
import { z } from "zod"

export const CriteriaSchema = z.object({
	nbWomens: z.coerce.number(),
	nbBachelors: z.coerce.number(),
	validCriteriaWoman: z.boolean(),
	validCriteriaBachelor: z.boolean()
})

export type Criteria = z.infer<typeof CriteriaSchema>;

File: backend/src/main/java/fr/eseo/tauri/model/Criteria.java
package fr.eseo.tauri.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.NonNull;

@Data
public class Criteria {

    @NonNull
    @JsonProperty
    private Integer nbWomens;

    @NonNull
    @JsonProperty
    private Integer nbBachelors;

    @NonNull
    @JsonProperty
    private Boolean validCriteriaWoman;

    @NonNull
    @JsonProperty
    private Boolean validCriteriaBachelor;

}